### PR TITLE
deletes auth token on signout

### PR DESCRIPTION
**Pull request by bergquist**
ref https://github.com/grafana/grafana/issues/15067

### CODE REVIEW

File: pkg/services/auth/auth_token_test.go [hunk 1]
```diff
@@ -46,6 +49,23 @@ func TestUserAuthToken(t *testing.T) {
...
 				So(LookupToken, ShouldBeNil)
 			})
+
+			Convey("signing out should delete token and cookie if present", func() {
+				token, err := userAuthTokenService.CreateToken(userID, "192.168.1.1:1234", "user agent")
+				So(err, ShouldBeNil)
+				So(token, ShouldNotBeNil)
+
+				httpreq := &http.Request{Header: make(http.Header)}
+				httpreq.AddCookie(&http.Cookie{Name: userAuthTokenService.Cfg.LoginCookieName, Value: token.AuthToken})
```
Comment (marefr) [inline pkg/services/auth/auth_token_test.go]: Probably need to use unhashed token as cookie value to resolve failing test.

Comment (bergquist) [reply pkg/services/auth/auth_token_test.go]: 🤦‍♂️

File: pkg/services/auth/auth_token.go [hunk 2]
```diff
@@ -111,8 +112,27 @@ func (s *UserAuthTokenServiceImpl) UserAuthenticatedHook(user *models.User, c *m
...
+
+	sql := `DELETE FROM user_auth_token WHERE auth_token = ?`
+	res, err := s.SQLStore.NewSession().Exec(sql, hashedToken)
+	if err != nil {
+		return err
+	}
+
+	affected, _ := res.RowsAffected()
+	if affected > 0 {
+		s.writeSessionCookie(c, "", -1)
```
Comment (marefr) [inline pkg/services/auth/auth_token.go]: I think we should always clear cookie on signout

**Review by torkelo — APPROVED** [review]
LGTM

**Review by xlson — APPROVED** [review]
LGTM

**Comment by torkelo** [comment]
Merge conflicts :(

**Pull request merged**